### PR TITLE
ULTIMA8: Expand use of Point3

### PR DESCRIPTION
**Pull request by OMGPizzaGuy**
Expanded use of Point3 throughout engine. This is a fairly large change, so I felt a review would be helpful.
There are a few bugs fixed in this found when reviewing previous changes. Thanks!

### CODE REVIEW

File: engines/ultima/ultima8/world/actors/attack_process.cpp [hunk 1]
```diff
@@ -281,23 +281,21 @@ void AttackProcess::run() {
...
 			waitFor(pid);
 			return;
 		}
 		case 0x8f:
 		{
 			// Pathfind to a point between npc and the target
-			int32 tx, ty, tz;
-			target->getLocation(tx, ty, tz);
-			int32 ax, ay, az;
-			target->getLocation(ax, ay, az);
```
Comment (OMGPizzaGuy) [inline engines/ultima/ultima8/world/actors/attack_process.cpp:297]: Bug in previous code?

File: engines/ultima/ultima8/world/item.cpp [hunk 2]
```diff
@@ -2271,21 +2276,20 @@ void Item::explode(int explosion_type, bool destroy_item, bool cause_damage) {
...
-							   160, false, xv, yv); //! CHECKME: 128?
+							   160, false, ptv.x, ptv.y); //! CHECKME: 128?
 
 		for (unsigned int i = 0; i < itemlist.getSize(); ++i) {
 			Item *item = getItem(itemlist.getuint16(i));
 			if (!item) continue;
 			if (getRange(*item, true) > 160) continue; // check vertical distance
 
-			item->getLocation(xv, yv, zv);
-			Direction dir = Direction_GetWorldDir(xv - xv, yv - yv, dirmode_8dirs); //!! CHECKME
```
Comment (OMGPizzaGuy) [inline engines/ultima/ultima8/world/item.cpp:2282]: Bug in previous code?

File: engines/ultima/ultima8/world/actors/attack_process.cpp (hunk 1 above)
Comment (mduggan) [reply engines/ultima/ultima8/world/actors/attack_process.cpp:297]: Sure looks like it!

File: engines/ultima/ultima8/world/actors/attack_process.cpp [hunk 3]
```diff
@@ -281,23 +281,21 @@ void AttackProcess::run() {
...
-			int32 x = (tx + ax) / 2;
-			int32 y = (ty + ay) / 2;
-			int32 z = (tz + az) / 2;
+			Point3 apt, tpt;
+			apt = a->getLocation();
+			tpt = target->getLocation();
```
Comment (bluegr) [inline engines/ultima/ultima8/world/actors/attack_process.cpp:295]:
```suggestion
			Point3 apt = a->getLocation();
			Point3 tpt = target->getLocation();
```

File: engines/ultima/ultima8/world/item.cpp (hunk 2 above)
Comment (mduggan) [reply engines/ultima/ultima8/world/item.cpp:2282]: Sure looks like it.

**Review by mduggan — APPROVED** [review]
Thanks for the huge cleanup! I haven't given it a playtest but I looked carefully through all the changes and couldn't see any issues.

I also noticed there are still a bunch of opportunities to do more (eg, passing x/y/z to various process constructors), but you have to cut it somewhere and this is alreay a big change so seems totally reasonable to merge.

**Pull request merged**